src/main/java/UseCases/CookbookUseCase.java

```diff
@@ -0,0 +1,51 @@
+package UseCases;
+
+import Commands.Implementations.CommandImpl;
+import Entities.Recipe;
+import Entities.Tag;
+import Storages.Implementations.RecipeStorageImpl;
+import Storages.Implementations.TagStorageImpl;
+
+import java.util.*;
+
+public class CookbookUseCase {
+    /**
+     * The cookbook displays recipes and shows recipes based on tags inputed into command
+     */
+    final RecipeStorageImpl recipeStorage;
+    final TagStorageImpl tagStorage;
+
+    public CookbookUseCase(RecipeStorageImpl recipes, TagStorageImpl tags) {
+        this.recipeStorage = recipes;
+        this.tagStorage = tags;
+    }
+
+    public RecipeResponseImpl run(CommandImpl command) {
+        List<Recipe> recipesMatched = new ArrayList<>();
+
+        if (!Objects.equals(command.get("FindAllRecipes"), "")) {
+            recipesMatched = new ArrayList<>(this.recipeStorage.recipes());
```

Comment:
We know all the values are strings, this `object.equals` is not needed. Instead, within the `if` try checking `command.containsKey` rather than checking if it's empty. Then just `get` it within the body of the `if`.